### ISSUE TITLE
German replacement is weird

### ISSUE THREAD

**Issue by jfinkhaeuser** [anyascii/anyascii#7]
As a native German with an Umlaut in my name, I'm really surprised by the choice to, as shown in the README, replace an Umlaut with the same base character. The "conventional" replacement is indeed a convention dating back to typewriters which did not originally contain Umlaut characters, so all Germans will recognize it as having the same meaning. Simply dropping the dots, however, changes pronunciation and potentially meaning.

Take the German word for bear, "Bär" as an example. I'm not saying it's the best example, but it's the first that comes to mind. Capitalization makes it somewhat distinguishable from "bar" (free from/of), except at the beginning of a sentence: here, the words are indistinguishable in your replacement scheme, and would be read as the latter word, but then conventional "Baer" would still retain its original sense.

I don't know what reasoning led to adopting this behaviour, but from a German point of view, it seems quite weird.

I also understand that from a non-German point of view, "Baer" is very difficult to know how to pronounce - but the spelling of "Bar" wouldn't result in an understandable pronunciation, either. My own name trips people up because it ends up with three vowels in a row. I do get the issue here, I just don't see how the adopted solution helps.

Hope that makes sense, and happy to discuss.

**Comment by hunterwb** [comment]
Diaresis/umlaut is used by dozens of languages and encoded identically so a language-neutral approach must be used. Adding the e is too disruptive for all text which is not German. If you know your input is German you may make this replacement yourself before calling anyascii.

**Comment by jfinkhaeuser** [comment]
Thank you for providing this context. 

Umlaut and diaresis are different things, closer to opposites really. I get your point about the encoding; there are ways to encode the two differently in Unicode, but I suppose most inputs won't do that. In such a case, you can't really know what the right thing to do is. When they're encoded differently, I hope that the replacement takes that into account - or is that too much to hope for?

I don't agree with the conclusion in the linked article, though. The missing "e" is very confusing to Germans - more so to Germans that are not used to going back and forth between it and English. I would go so far as to suggest that this reads very clearly like a non-expert opinion (which the author all but admits he is).

Still, this has brought to light the problem you're facing.

Maybe the right conclusion is that the one-line description "Converts Unicode characters to their best ASCII representation" is wrong. I think there's great value in what the library does - this description, though, assumes a very particular interpretation of the word "best" that I'm certain not everybody shares.

How about changing it to a "Converts Unicode characters to a simple, readable ASCII representation without considering context or language"? It seems to avoid creating false expectations much better.

**Comment by bmuessig** [comment]
> How about changing it to a "Converts Unicode characters to a simple, readable ASCII representation without considering context or language"? It seems to avoid creating false expectations much better.

Agreed. It's not as production-ready as I had hoped, but still very useful. Thanks @hunterwb for all your effort!

**Comment by CharlesTerrell** [comment]
From looking at the table, the strange mapping is not exclusive to German.

Spanish:  Ñ/ñ should be ny, not just n.  Very obvious.  Discarding the tilde is usually understandable in context, but could lead to misunderstandings (or at least.eye-rolling).

Esperanto:  Ĉ/Ĵ/Ĝ/Ĥ/Ŝ/Ŭ can be converted with one of two systems.  Either replace the letter with its bare form followed by x (Ĝ->Gx, ŭ->ux), or follow with h except for the ŭ (Ĝ->Gh, ŭ->u).  Both systems have good and bad points.  But simply using the bare Roman letter is almost always wrong, and usually changes the meaning in hilarious ways.

The table might need to be modified for specific use.cases.  Which somewhat limits its universality.
